org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/GridView.java

```diff
@@ -662,7 +768,7 @@ private void setupColumns()
 		
 		listbox.appendChild(columns);
 		columns.setSizable(true);
-		columns.setMenupopup("none");
+		//columns.setMenupopup("none");
```

Comment:
Please just remove what's no longer needed, keep it clean.

Reply:
Why are we not using columns.setMenupopup here instead of the getRoot() attribute approach above ? setMenupopup is the recommended Zk approach and we must have a good reason to use a different approach (the root attribute approach does looks more like a hack here).

Reply:
Tried that method first, but the menu didn't appear.
 To troubleshoot, I recreated the scenario on [zkfiddle.org](https://zkfiddle.org/sample/gig0j7/5-Customized-Menupopup-On-Grid-Header), which uses the EE editor, and it functioned correctly there.
Returning to idempiere, I retried setting the menu popup to 'auto,' and it worked as expected

The ability to customize the context menu popup appears to be malfunctioning within the CE editor

Reply:
ok, will check it out.

Reply:
hi @hieplq , yes, you are right, custom menu popup only works in EE.

Anyway, I would prefer the following that's quite a bit simpler:

[pr2355-menupopup.patch.txt](https://github.com/idempiere/idempiere/files/15258710/pr2355-menupopup.patch.txt)